src/test/java/ca/corefacility/bioinformatics/irida/ria/unit/web/services/UIProjectSampleServiceTest.java

```diff
@@ -75,4 +81,24 @@ public void testCreateSample() {
 		ResponseEntity<AjaxResponse> response = service.createSample(request, PROJECT_1_ID, Locale.ENGLISH);
 		assertEquals(HttpStatus.OK, response.getStatusCode(), "Sample should be created");
 	}
+
+	@Test
+	public void testCreateSampleWithMetadata() {
+		List<FieldUpdate> metadata = new ArrayList<>();
+		metadata.add(new FieldUpdate("field1", "value1"));
+		metadata.add(new FieldUpdate("field2", "value2"));
+		CreateSampleRequest request = new CreateSampleRequest(GOOD_NAME, null, null, metadata);
+		ResponseEntity<AjaxResponse> response = service.createSample(request, PROJECT_1_ID, Locale.ENGLISH);
+		assertEquals(HttpStatus.OK, response.getStatusCode(), "Sample should be created");
+	}
+
+	@Test
+	public void testUpdateSampleWithMetadata() {
+		List<FieldUpdate> metadata = new ArrayList<>();
+		metadata.add(new FieldUpdate("field1", "value1"));
+		metadata.add(new FieldUpdate("field2", "value2"));
+		CreateSampleRequest request = new CreateSampleRequest(GOOD_NAME, null, null, metadata);
```

Comment:
If you set name as Null does the update still work as expected?

Reply:
I check if the new value is empty or null before trying to update the sample. Setting the name to null will not update the sample, but sample names have a null constraint. But I am wondering now, if should remove the null or empty check for description, because it doesn't have a null constraint. 

![image](https://user-images.githubusercontent.com/325703/188727851-f92a4a00-0782-4fe6-8ef7-9431e89ec22f.png)

Reply:
The sample service is mocked for these unit tests. Testing the sample name is null would probably get tested in SampleServiceImplIT.java. Unless I should unmock them??? If an error is thrown, I do return an AjaxErrorResponse now and I made sure if either updating the metadata or sample fails it reverts. It's a little tricky implementing an endpoint that's not being used at the moment, lol.

Reply:
Could we make an `UpdateSampleRequest` class that is just an extended version of the `CreateSampleRequest` class with a new class description. That way it makes a litte bit more sense to the developer even though they are the same format.

Reply:
Please see 94c3cbccf0d86bca32ff786dadb5de0207272441.